### PR TITLE
test(coverage): bring updater/ and watcher/ to 80%

### LINKED ISSUE #1021: coverage: bring updater/ and watcher/ modules to 80%
## Problem

Two critical modules have minimal effective test coverage despite having test files:

### updater/ (7 modules, 6 test files)
- `background.py`, `checker.py`, `installer.py`, `manager.py`, `sidecar_updater.py`, `state.py`
- Handles **security-critical auto-update functionality**
- Tests exist in `tests/updater/` but coverage is near 0%

### watcher/ (5 modules, 4 test files + 1 integration)
- `config.py`, `handler.py`, `monitor.py`, `queue.py`
- Handles **file system monitoring** — bugs here cause silent data loss
- Tests exist in `tests/watcher/` and `tests/integration/test_watcher_components.py`

## Why This Matters

Both modules interact with the OS (filesystem, subprocess) and have failure modes that are invisible without tests. The updater module in particular is a supply-chain attack surface.

## Acceptance Criteria

- [ ] `updater/` module coverage ≥ 80%
- [ ] `watcher/` module coverage ≥ 80%
- [ ] Error paths (network failures, permission errors, corrupt state) tested
- [ ] Integration tests for updater state persistence and watcher event handling

## Measurement

```bash
python -m pytest tests/updater/ tests/watcher/ --cov=file_organizer.updater --cov=file_organizer.watcher --cov-report=term-missing
```

### PR DESCRIPTION
**Pull request by curdriceaurora**
## Summary
- Raise updater/ module coverage to 99-100% (from ~85-98%) across all 6 modules
- Raise watcher/ module coverage to 99-100% (from ~96-100%) across all 4 modules
- Key fix: `PYTEST_CURRENT_TEST` env var was causing early returns in `background.py`, masking untested paths
- 548 lines of new tests covering error paths (rollback failures, checksum edge cases, download failures)

Closes #1021

## Test plan
- [x] `python -m pytest tests/updater/ tests/watcher/ --cov=file_organizer.updater --cov=file_organizer.watcher` shows ≥80% for all modules
- [x] Pre-commit hooks pass

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Tests**
  * Expanded update-system tests: added coverage for coordinated update failures, download/verification errors, checksum handling, installer rollbacks, and cleanup after replace failures.
  * Added version-detection and HTTP-failure handling tests for the checker.
  * Improved background-check tests to consistently isolate environment variables.
  * Expanded file-monitor tests for callback registration and directory add/remove edge cases.

### CODE REVIEW

**Review by curdriceaurora — COMMENTED** [review]
## 🤖 Auto Claude PR Review

### Merge Verdict: 🟠 NEEDS REVISION

> **🟠 Needs revision** - 2 issue(s) require attention.

2 issue(s) must be addressed (0 required, 2 recommended), 2 suggestions

### Risk Assessment
| Factor | Level | Notes |
|--------|-------|-------|
| Complexity | High | Based on lines changed |
| Security Impact | None | Based on security findings |
| Scope Coherence | Good | Based on structural review |

### Findings Summary
- **Medium**: 2 issue(s)
- **Low**: 2 issue(s)

---
_Generated by Auto Claude PR Review_

### Findings (4 selected of 4 total)

#### 🟡 [e3a7144ee217] [MEDIUM] Duplicated PYTEST_CURRENT_TEST env removal pattern across 8 tests
📁 `tests/updater/test_background.py:44`

The same 7-line pattern (env_patch.start(), try/if PYTEST_CURRENT_TEST/del, finally/env_patch.stop()) is repeated in 8 out of 10 test methods in this file (lines 45-57, 64-80, 88-115, 122-152, 159-187, 194-213, 220-236, 245-273). This is a textbook case where a pytest fixture should be used instead. A fixture like `@pytest.fixture(autouse=True) def clear_pytest_env()` that yields with PYTEST_CURRENT_TEST removed would eliminate ~56 lines of boilerplate and significantly improve readability. Additionally, the manual env_patch.start()/stop() with try/finally is fragile compared to a context manager or fixture approach.

**Suggested fix:**
```
Extract a pytest fixture:

@pytest.fixture(autouse=True)
def _clear_pytest_env():
    """Remove PYTEST_CURRENT_TEST so maybe_check_for_updates proceeds past the env guard."""
    old = os.environ.pop('PYTEST_CURRENT_TEST', None)
    yield
    if old is not None:
        os.environ['PYTEST_CURRENT_TEST'] = old

Then remove the env_patch/try/finally boilerplate from all 8 tests. The two tests that explicitly test the PYTEST_CURRENT_TEST guard behavior (test_returns_none_when_in_pytest, test_returns_none_when_disabled_via_env) can override this fixture as needed.
```

#### 🔵 [03fda8900568] [LOW] Test couples to internal _on_moved_callbacks attribute
📁 `tests/watcher/test_monitor.py:461`

The test `test_on_moved_callback_registers` asserts on the handler's internal `_on_moved_callbacks` list (`assert _cb in monitor.handler._on_moved_callbacks`). This couples the test to the handler's private implementation. If the handler's internal callback storage changes (e.g., switching to a dict, set, or observer pattern), this test breaks even if the public API still works. Other callback tests in this file (test_on_created_callback, test_on_deleted_callback, test_on_modified_callback) correctly test through the public interface by triggering real file events and verifying the callbacks fire.

**Suggested fix:**
```
Consider testing the on_moved callback by performing a real file rename (shutil.move) and verifying the callback fires, similar to the pattern used for on_created/on_deleted/on_modified tests in the TestFileMonitorCallbacks class. This would test the actual behavior rather than implementation details.
```

#### 🟡 [cbc323907105] [MEDIUM] New test classes in test_monitor.py missing @pytest.mark.unit
📁 `tests/watcher/test_monitor.py:449`

Three new test classes (`TestFileMonitorOnMovedCallback`, `TestFileMonitorRemoveDirectoryEdgeCases`, `TestFileMonitorAddDirectoryEdgeCases`) lack the `@pytest.mark.unit` decorator. All four existing primary test classes in this file use it (`TestFileMonitorLifecycle` at line 98, `TestFileMonitorFileDetection` at line 139, `TestFileMonitorDynamicDirectories` at line 313, `TestFileMonitorCallbacks` at line 378). Unlike other test files (e.g., test_background.py, test_installer.py, test_state.py), test_monitor.py has NO module-level `pytestmark`, so the class-level decorator is the only way these tests get the `unit` marker. Without it, running `pytest -m unit` will skip these tests. Searched `Grep('pytestmark', 'tests/watcher/test_monitor.py')` - confirmed no module-level pytestmark exists.

**Suggested fix:**
```
Add `@pytest.mark.unit` decorator before each of the three new test classes (lines 449, 464, 501) to match the convention used by all other test classes in this file.
```

#### 🔵 [33f6f76dad5b] [LOW] In-function imports of private helpers instead of top-level imports
📁 `tests/updater/test_installer.py:441`

TestPlatformHints and TestScoreAsset classes import `_get_platform_hints`, `_get_arch_hints`, and `_score_asset` inside individual test methods rather than at the top of the file. The codebase convention is to import even private/underscore-prefixed functions at the top level (e.g., `test_updater_checker.py` line 17 imports `_parse_version` at the top; `tests/watcher/test_config.py` imports `_matches_pattern` at the top). Searched `Grep('from file_organizer.*import.*_', 'tests/')` - confirmed multiple private function imports at top level across the test suite.

**Suggested fix:**
```
Move `from file_organizer.updater.installer import _get_platform_hints, _get_arch_hints, _score_asset` to the top-level imports alongside `InstallResult, UpdateInstaller`, and remove the in-function imports. Similarly, move `import sys` (line 69) to the stdlib imports section.
```

---
*This review was generated by Auto Claude.*

File: tests/updater/test_background.py [hunk 1]
```diff
@@ -42,26 +42,42 @@ def test_returns_none_when_in_pytest(self):
...
+        env_patch.start()
+        try:
+            if "PYTEST_CURRENT_TEST" in os.environ:
+                del os.environ["PYTEST_CURRENT_TEST"]
+            with patch("file_organizer.updater.background.ConfigManager") as mock_cfg_mgr:
+                mock_cfg = MagicMock()
+                mock_cfg.updates.check_on_startup = False
+                mock_cfg_mgr.return_value.load.return_value = mock_cfg
+                result = maybe_check_for_updates()
+                assert result is None
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply tests/updater/test_background.py]: Fixed in 174e19d9. The updater tests now clear both env gates and assert the collaborator interactions so the intended None-return branches are proven, not just the final outcome.

File: tests/watcher/test_monitor.py [hunk 2]
```diff
@@ -446,6 +446,82 @@ def _cb(e: object) -> None:
...
+            debounce_seconds=0.0,
+        )
+        mon = FileMonitor(config)
+        # We need a new dir that resolves to a different path
+        new_dir = tmp_path / "new_dir"
+        new_dir.mkdir()
+        mon.add_directory(new_dir)
+        # new_dir should be appended
+        assert new_dir.resolve() in [d.resolve() for d in mon.config.watch_directories]
+
```
[the remark this answers is not in the document]
Comment (curdriceaurora) [reply tests/watcher/test_monitor.py]: Fixed in 174e19d9. The watcher test now re-adds the same configured path and asserts the directory list is unchanged, so it exercises the already-in-config branch it claims to cover.
